src/vs/workbench/services/workingCopy/common/storedFileWorkingCopy.ts

```diff
@@ -1019,7 +1019,11 @@ export class StoredFileWorkingCopy<M extends IStoredFileWorkingCopyModel> extend
 
 					// Delegate to working copy model save method if any
 					if (typeof resolvedFileWorkingCopy.model.save === 'function') {
-						stat = await resolvedFileWorkingCopy.model.save(writeFileOptions, saveCancellation.token);
+						const result = await resolvedFileWorkingCopy.model.save(writeFileOptions, saveCancellation.token);
```

Comment:
This is weird to me: the API allows me to return `undefined` but then we throw if that is the case unless cancelled?

Maybe we come up with a new return type that is wrapping around a `Promise` and cancellation? There is `CancelablePromise` but it seems to miss a property if its canceled or not.

Reply:
CancelablePromise would not work well for this - we want to make sure that we wait for the promise to completion:
If we cancelled and it finished anyway, we need to process the result.
If it doesn't complete the task (returns undefined) but we didn't ask to cancel, then something has gone wrong and it should throw.

So, a new wrapper type would just be using this same logic as far as I can see

Reply:
I modified it to use a cancellation exception instead of an alternate return type.